openai-assistant/src/com/axonivy/connector/openai/assistant/ui/CopyableInfoDialog.java

```diff
@@ -0,0 +1,137 @@
+  protected void configureShell(Shell newShell) {
+    super.configureShell(newShell);
+    newShell.setText(title);
+    newShell.setMinimumSize(750, 400);
+  }
+
+  @Override
+  protected Point getInitialSize() {
+    return new Point(750, 400);
+  }
+
+  @Override
+  protected Point getInitialLocation(Point initialSize) {
+    Rectangle screenSize = getShell().getDisplay().getPrimaryMonitor().getBounds();
+
+    // Calculate the center position
+    int x = (screenSize.width - initialSize.x) / 2;
+    int y = (screenSize.height - initialSize.y) / 2;
+
+    return new Point(x, y);
+  }
+
+  @Override
+  protected boolean isResizable() {
+    return true;
+  }
+
+  @Override
+  protected Control createDialogArea(Composite parent) {
+    Composite container = (Composite) super.createDialogArea(parent);
+    container.setLayout(new GridLayout(1, false));
+
+    browser = new Browser(container, SWT.BORDER);
+    browser.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+    Parser parser = Parser.builder().build();
+    HtmlRenderer renderer = HtmlRenderer.builder().build();
+    String htmlContent = renderer.render(parser.parse(this.message));
+
+    // Get Eclipse theme colors
+    Color bgColor = JFaceResources.getColorRegistry().get(JFacePreferences.CONTENT_ASSIST_BACKGROUND_COLOR);
+    Color fgColor = JFaceResources.getColorRegistry().get(JFacePreferences.CONTENT_ASSIST_FOREGROUND_COLOR);
+
+    // Create CSS with Eclipse colors
+    String css = String.format("body { background-color: %s; color: %s; font-family: '%s'; }", toHex(bgColor.getRGB()),
+        toHex(fgColor.getRGB()), JFaceResources.getTextFont().getFontData()[0].getName());
+
+    // Wrap HTML content with custom CSS
+    String htmlWithCss = String.format("<html><head><style>%s</style></head><body>%s</body></html>", css, htmlContent);
+
+    browser.setText(htmlWithCss);
+
+    browser.addProgressListener(new ProgressListener() {
+      @Override
+      public void completed(ProgressEvent event) {
+        getShell().getDisplay().asyncExec(() -> resizeDialog());
+      }
+
+      @Override
+      public void changed(ProgressEvent event) {
+        // Not needed for this implementation
+      }
+    });
+
+    return container;
+  }
+
+  @Override
+  protected void createButtonsForButtonBar(Composite parent) {
+    createButton(parent, 1, "Copy to Clipboard", false);
+    createButton(parent, OK, "OK", true);
+  }
+
+  @Override
+  protected void buttonPressed(int buttonId) {
+    if (buttonId == 1) {
+      copyToClipboard();
+    } else {
+      super.buttonPressed(buttonId);
+    }
+  }
+
+  private void copyToClipboard() {
+    Clipboard clipboard = new Clipboard(getShell().getDisplay());
+    TextTransfer textTransfer = TextTransfer.getInstance();
+    clipboard.setContents(new Object[] { message }, new Transfer[] { textTransfer });
+    clipboard.dispose();
+  }
+
+  private String toHex(RGB rgb) {
+    return String.format("#%02x%02x%02x", rgb.red, rgb.green, rgb.blue);
```

Comment:
Could we create a constant for these format since its name can somehow explain the pattern & easy to maintain/refactor for later?